src/PersistStore.ts

```diff
@@ -140,17 +142,17 @@ export class PersistStore<T, P extends keyof T> {
 
     this.cancelWatch = reaction(
       () => {
-        const propertiesToWatch: Record<string, unknown> = {};
+        const propertiesToWatch = <Record<P, unknown>>{};
```

Comment:
This line seems wrong. Should it be the old way?

```suggestion
        const propertiesToWatch: Record<P, unknown> = {};
```

Reply:
It doesn't work. If you use this way, typescript will try to initialize object with keys from generic value and it will be an error for empty objects. 

Anyway. If you want to initialize empty object with generic value in record type you need to use my suggestion OR use `as`. 

```ts
type A = { B: string }

const r1: Record<string, string> = {} // OK
const r2: Record<keyof A, string> = {} // Property 'B' is missing in type '{}' but required in type 'Record<"B", string>'
const r3 = <Record<keyof A, string>>{} // OK
const r4 = {} as Record<keyof A, string> // OK
```

Reply:
@codeBelt what about this?

Reply:
@quarrant I am not sure. Go with what you think is best. I like `r4` better than `r3`.

Reply:
agree